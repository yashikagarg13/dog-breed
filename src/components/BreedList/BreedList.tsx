import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Chip from '@material-ui/core/Chip'
import CloseIcon from '@material-ui/icons/Close'
import {NO_BREEDS_SELECTED} from '../../constants/messages'

type BreedListProps = {
  breeds: string[];
  showNoBreedMessage?: boolean;
  onDelete: Function;
}
const useStyles = makeStyles(theme => ({
  breedContainer: {
    display: 'flex',
    alignItems: 'center',
    flexWrap: 'wrap',
    gap: 15,
  },
  chip: {
    textTransform: 'capitalize',
    height: 36,
    borderRadius: 6,
    color: '#546079',
    background: '#E9E9ED',
    fontSize: 14,
    fontWeight: 'bold',
    '&:hover': {
      backgroundColor: '#cdcdcd'
    }
  },
  deleteIcon: {
    color: '#546079',
    fontSize: 16,
    margin: '0 5px  0 6px',
  }
}))
function BreedList(props: BreedListProps) {
  const classes = useStyles()
  const { breeds, showNoBreedMessage, onDelete } = props

  return (
    <div className={classes.breedContainer}>
      {(breeds.length === 0 && showNoBreedMessage) && (
        <p>{NO_BREEDS_SELECTED}</p>
      )}
      {(breeds).map(breed => <Chip 
        key={breed} 
        label={breed} 
        onDelete={() => onDelete(breed)} 
        className={classes.chip} 
        classes={{ deleteIcon: classes.deleteIcon }}
        deleteIcon={<CloseIcon data-testid={`${breed}-close-icon`} />}
      />)}
    </div>
  )
}

export default BreedList