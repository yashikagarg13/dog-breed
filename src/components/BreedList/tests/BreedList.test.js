import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import BreedList from '../BreedList'

describe('BreedList Component', () => {
  let props
  const breeds = ['hound', 'australian']

  beforeEach(() => {
    props = {
      breeds, 
      showNoBreedMessage: false, 
      onDelete: jest.fn()
    }
  })

  it('should render BreedList component', () => {
    const { container } = render(
      <BreedList {...props} />
    )
    expect(container).toBeInTheDocument()
    expect(container.innerHTML).toMatchSnapshot()
  })

  it('should show no breeds message id showNoBreedMessage is true and no  breeds selected', () => {
    props.breeds = []
    props.showNoBreedMessage = true
    const { container } = render(
      <BreedList {...props} />
    )
    expect(container).toBeInTheDocument()
    expect(container.innerHTML).toMatchSnapshot()
  })

  it('should call onDelete on clicking delete icon in breed chip', () => {
    const { container, getByTestId } = render(
      <BreedList {...props} />
    )
    const closeBtn = getByTestId('hound-close-icon')
    fireEvent.click(closeBtn)

    expect(props.onDelete).toHaveBeenCalledWith('hound')
  })
})
