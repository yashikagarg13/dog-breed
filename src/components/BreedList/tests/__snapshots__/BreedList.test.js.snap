// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedList Component should render BreedList component 1`] = `"<div class=\\"makeStyles-breedContainer-1\\"><div role=\\"button\\" class=\\"MuiChip-root makeStyles-chip-2 MuiChip-deletable\\" tabindex=\\"0\\"><span class=\\"MuiChip-label\\">hound</span><svg class=\\"MuiSvgIcon-root MuiChip-deleteIcon makeStyles-deleteIcon-3\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"hound-close-icon\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></div><div role=\\"button\\" class=\\"MuiChip-root makeStyles-chip-2 MuiChip-deletable\\" tabindex=\\"0\\"><span class=\\"MuiChip-label\\">australian</span><svg class=\\"MuiSvgIcon-root MuiChip-deleteIcon makeStyles-deleteIcon-3\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"australian-close-icon\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></div></div>"`;

exports[`BreedList Component should show no breeds message id showNoBreedMessage is true and no  breeds selected 1`] = `"<div class=\\"makeStyles-breedContainer-4\\"><p>No breeds selected. Click on \\"Manage Breeds\\" to find some</p></div>"`;
